src/common/state/ColorControlPointList.C

```diff
@@ -1785,20 +1795,56 @@ ColorControlPointList::CompactCreateNode(DataNode *parentNode, bool completeSave
 //
 // Programmer: Justin Privitera
 // Creation:   Fri Jun  3 11:27:43 PDT 2022
+// 
+// Note: We assume the following: the tag is not an empty string AND the tag 
+//   contains only alphanumeric characters and characters in " -=<>". In other
+//   words, we assume that the result of `ValidateTag` is true.
 //
 // Modifications:
 //    Justin Privitera, Wed Jun 29 17:50:24 PDT 2022
 //    Set tagChangesMade to true every time a tag is added.
+// 
+//    Justin Privitera, Thu Aug 25 15:04:55 PDT 2022
+//    Now takes a const string and uses `emplace_back`.
 //
 // ****************************************************************************
 
 void
-ColorControlPointList::AddTag(std::string newtag)
+ColorControlPointList::AddTag(const std::string newtag)
 {
     // If the tag is already in the tag list then we will do nothing.
     if (!HasTag(newtag))
     {
-        tagNames.push_back(newtag);
+        tagNames.emplace_back(newtag);
+        tagChangesMade = true;
+    }
+}
+
+// ****************************************************************************
+// Method: ColorControlPointList::RemoveTag
+//
+// Purpose:
+//   Remove the tag from the list of local tags.
+//
+// Programmer: Justin Privitera
+// Creation:   Wed Aug 10 15:35:58 PDT 2022
+//
+// Modifications:
+//    Justin Privitera, Thu Aug 25 15:04:55 PDT 2022
+//    Made the "tag" arg const.
+//
+// ****************************************************************************
+
+void
+ColorControlPointList::RemoveTag(const std::string tag)
+{
+    // If the tag is not in the tag list then we will do nothing.
+    int index = GetTagIndex(tag);
+    if (index != -1)
+    {
+        stringVector::iterator pos = tagNames.begin();
+        for (int i = 0; i < index; i ++) pos ++;
```

Comment:
We generally put the code block on a separate line from the for-statement.
Similar for line below with if-statement.

Reply:
I'll change this